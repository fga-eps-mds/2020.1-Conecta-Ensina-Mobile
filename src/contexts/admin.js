import React, {createContext, useState, useContext, useEffect} from 'react';
import {AuthContext} from './auth';

export const AdmContext = createContext({});

function AdmProvider({children}) {
  const [students, setStudents] = useState(null);
  const [userDB, setUserDB] = useState(null);
  const [teacher, setTeacher] = useState(null);

  const {Host, user} = useContext(AuthContext);

  async function getProfessorList() {
    const fetchResponse = await fetch(`${Host}/api/student/status/0`);

    console.log(Host);

    try {
      const data = await fetchResponse.json();
      setStudents(data.data.student);
    } catch (error) {
      return error;
    }
  }

  async function statusUpdate(id) {
    const settings = {
      method: 'PUT',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        status: 1,
        agentRole: user.role,
      }),
    };
    const fetchResponse1 = await fetch(
      `${Host}/api/student/status/${id}`,
      settings,
    );
    try {
      const data = await fetchResponse1.json();
      console.log('Success:', data);
    } catch (error) {
      console.error('Error:', error);
    }
  }

  async function getProfessoUser(id) {
    const fetchResponse = await fetch(`${Host}/api/user/${id}`);
    try {
      const data = await fetchResponse.json();
      let usuario = {
        id: id,
        firstName: data.data.user.firstName,
        lastName: data.data.user.lastName,
        email: data.data.user.email,
        password: data.data.user.password,
        role: data.data.user.role,
        cellphone: data.data.user.cellphone,
      };
      setUserDB(usuario);
      try {
        let response = await fetch(`${Host}/api/teacher/${id}`);
        const data = await response.json();
        let professor = {
          photo: data.data.teacher.photo,
          video: data.data.teacher.video,
          graduation_area: data.data.teacher.graduation_area,
          degree: data.data.teacher.degree,
          bank: data.data.teacher.bank,
          agency: data.data.teacher.agency,
          account: data.data.teacher.account,
        };
        setTeacher(professor);
      } catch (error) {
        console.error(error);
      }
    } catch (error) {
      return error;
    }
  }

  return (
    <AdmContext.Provider
      value={{
        students,
        userDB,
        teacher,
        getProfessorList,
        statusUpdate,
        getProfessoUser,
      }}>
      {children}
    </AdmContext.Provider>
  );
}

export default AdmProvider;
