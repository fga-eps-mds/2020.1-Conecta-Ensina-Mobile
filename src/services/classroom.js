export async function getNextClassroom(Host) {
  const fetchResponse = await fetch(
    `${Host}/api/classroom/nextClass/3bd7c190-ce64-4827-8c0c-58cfef45ad9f`,
  );
  try {
    const data = await fetchResponse.json();
    return data.data.classroom;
  } catch (error) {
    return error;
  }
}

export async function getUserClassroom(Host, id) {
  const fetchResponse = await fetch(`${Host}/api/classroom/userClasses/${id}`);
  try {
    const data = await fetchResponse.json();
    return data.data.classroom;
  } catch (error) {
    return error;
  }
}

export async function createClass(user, teacher, filter, student, Host) {
  const response = await fetch(`${Host}/api/classroom/create`, {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      agentRole: user.role,
      teacher: teacher,
      student: user.id,
      grade: student.grade,
      subject: filter.subject,
      dtclass: filter.dtClass,
      duration: filter.duration,
      cep: student.cep,
      number: student.number,
      details: 'oi',
    }),
  });
  try {
    const data = await response.json();
    console.log('Success: ', data);
    return data;
  } catch (error) {
    console.log(error);
  }

  console.log(response);
}
