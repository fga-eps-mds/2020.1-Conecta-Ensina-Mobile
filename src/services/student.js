export async function getStudent(Host, id) {
  const response = await fetch(`${Host}/api/student/${id}`);
  try {
    const data = await response.json();
    console.log(data.data.student);
    //setSubjects(data.data.subject);
    return data.data.student;
  } catch (error) {
    return error;
  }
}
