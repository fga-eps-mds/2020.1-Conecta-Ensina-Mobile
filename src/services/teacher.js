export async function getAllTeachers(Host) {
  const fetchResponse = await fetch(`${Host}/api/teacher/`);
  try {
    const data = await fetchResponse.json();
    console.log(data.data.user);
    //setTeachers(data.data.user);
    return data.data.user;
  } catch (error) {
    return error;
  }
}
