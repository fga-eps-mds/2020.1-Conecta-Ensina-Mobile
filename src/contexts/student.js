import React, {createContext, useState, useContext} from 'react';
import {AuthContext} from './auth';
import * as Student from '../services/student';

export const StudentContext = createContext({});

export default function UserProvider({children}) {
  const [student, setStudent] = useState({});
  const {Host} = useContext(AuthContext);

  async function getStudent(id) {
    const response = await Student.getStudent(Host, id);
    if (student !== response) {
      console.log(response);
      setStudent(response);
    }
  }

  return (
    <StudentContext.Provider value={{student, getStudent}}>
      {children}
    </StudentContext.Provider>
  );
}
