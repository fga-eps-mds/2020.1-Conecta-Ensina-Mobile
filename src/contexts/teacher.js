import React, {createContext, useState, useContext} from 'react';
import {AuthContext} from './auth';
import * as Teacher from '../services/teacher';

export const TeacherContext = createContext({});

export default function TeacherProvider({children}) {
  const [teacher, setTeacher] = useState({});
  const {Host} = useContext(AuthContext);

  async function loadTeachers() {
    const response = await Teacher.getAllTeachers(Host);
    if (teacher !== response) {
      console.log(response);
      setTeacher(response);
    }
  }
  async function getTeacher(id) {
    const response = await Teacher.getTeacher(Host, id);
    if (teacher !== response) {
      console.log(response);
      setTeacher(response);
    }
  }

  return (
    <TeacherContext.Provider value={{teacher, loadTeachers, getTeacher}}>
      {children}
    </TeacherContext.Provider>
  );
}
