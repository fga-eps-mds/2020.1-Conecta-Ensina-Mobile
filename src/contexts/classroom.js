import React, {createContext, useState, useContext} from 'react';
import {AuthContext} from './auth';
import * as Class from '../services/classroom';
import * as Student from '../services/student';
import * as Address from '../services/findAddress';
import * as GeoCode from '../services/geocode';
import {FiltersContext} from '../contexts/filters';
import {StudentContext} from '../contexts/student';

export const ClassroomContext = createContext({});

export default function ClassroomProvider({children}) {
  const [classroom, setClassroom] = useState({});
  const [firstClass, setFirstClass] = useState(null);
  const [classes, setClasses] = useState(null);
  const {Host, user} = useContext(AuthContext);
  const {stackStudents} = useContext(StudentContext);
  const {filter} = useContext(FiltersContext);
  const [statusClasses, setStatusClasses] = useState([]);
  const [statusClass, setStatusClass] = useState({});
  const [coordenates, setCoordenates] = useState({});

  async function loadNextClass() {
    const response = await Class.getNextClassroom(Host);
    if (classroom !== response) {
      setFirstClass(response);
    }
  }
  async function createClass(teacher) {
    //console.log('Esse aqui Ã© o professor: ' + teacher);
    const student = await Student.getStudent(Host, user.id);
    const response = await Class.createClass(
      user,
      teacher,
      filter,
      student.student,
      Host,
    );
  }

  async function loadUserClasses() {
    const response = await Class.getUserClassroom(Host, user.id);
    if (classroom !== response) {
      setClassroom(response);
    }
  }

  async function getClassroom(status) {
    let teacher, student;
    if (user.role === 2) {
      student = '';
      teacher = user.id;
    } else {
      student = user.id;
      teacher = '';
    }
    const response = await Class.getClassroom(student, teacher, status, Host);

    if (classroom !== response) {
      //console.log(response);
      setStatusClasses(response);
    }
  }

  async function updateStatusClassroom(id) {
    let response = await Class.readClass(Host, id);
    //console.log(response.status);
    if (response.status === 1) {
      response = await Class.updateStatusClassroom(id, 2, Host);
      //console.log(response.message);
    } else if (response.status === 2) {
      response = await Class.updateStatusClassroom(id, 3, Host);
      //console.log(response.message);
    } else if (response.status === 3) {
      response = await Class.updateStatusClassroom(id, 4, Host);
      //console.log(response.message);
    } else if (response.status === 4) {
      response = await Class.updateStatusClassroom(id, 5, Host);
      //console.log(response.message);
    }
  }
  async function loadStatusClasses(status) {
    const response = await Class.getStatusClassroom(Host, user.id, status);
    if (statusClasses !== response) {
      await setStatusClasses(response);
    }
    //console.log(response);
    return response;
  }

  async function loadStatusClassesStudents(status) {
    const response = await loadStatusClasses(status);
    let studentList = await {list: []};
    await Promise.all(
      response.map(async (item) => {
        const student = await Student.getStudent(Host, item.student);
        studentList.list = await studentList.list.concat(student);
      }),
    );
    await stackStudents(studentList.list);
  }

  async function readClass(id) {
    const response = await Class.readClass(Host, id);
    const address = await Address.findAddress(response.cep);
    let responseClass = {
      id: response.role,
      status: response.status,
      teacher: response.teacher,
      student: response.student,
      grade: response.grade,
      subject: response.subject,
      dtclass: response.dtClass,
      duration: response.duration,
      cep: response.cep,
      number: response.number,
      details: response.details,
      address: address,
    };

    setClassroom(responseClass);
  }

  async function getClass() {
    const response = await Class.getClass(Host, user);
    setClasses(response);
  }

  async function updateStatusClasses(id, status) {
    const response = await Class.updateStatus(id, status, Host);
    if (classroom !== response) {
      setClassroom(response);
    }
  }

  async function geoCode(cep) {
    const response = await GeoCode.geocode(cep);
    if (coordenates !== response) {
      //console.log(response.results[0].geometry.location);
      setCoordenates(response.results[0].geometry.location);
    }
  }

  return (
    <ClassroomContext.Provider
      value={{
        classroom,
        firstClass,
        classes,
        loadNextClass,
        createClass,
        loadUserClasses,
        getClassroom,
        updateStatusClassroom,
        loadStatusClasses,
        readClass,
        statusClasses,
        statusClass,
        setStatusClass,
        setStatusClasses,
        Host,
        getClass,
        updateStatusClasses,
        loadStatusClassesStudents,
        geoCode,
        coordenates,
      }}>
      {children}
    </ClassroomContext.Provider>
  );
}
