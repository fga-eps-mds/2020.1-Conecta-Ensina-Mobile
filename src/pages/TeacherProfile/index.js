import React, {useContext, useEffect, useState} from 'react';
import Theme from '../../../Theme';
import {ClassroomContext} from '../../contexts/classroom';
import {TeacherContext} from '../../contexts/teacher';
import Background2 from '../../components/Background2';
import ContinueContainer from '../../components/ContinueContainer';
import RedContainerText from '../../components/RedContainerText';
import CustomTextContainer from '../../components/CustomTextContainer';
import gradeResolver from '../../services/gradeResolver';
import {
  ContainerB,
  ContainerW,
  Icon,
  Logo,
  UserContatiner,
  ButtonContainer,
  ComplainButton,
} from './styles';
import CustomText from '../../components/CustomText';
import {StudentContext} from '../../contexts/student';

export default function TeacherProfile({navigation, route}) {
  const {teacher} = useContext(TeacherContext);
  const {classroom, createClass} = useContext(ClassroomContext);
  const {getStudent} = useContext(StudentContext);

  useEffect(() => {
    //console.log(teacher);
  }, [teacher]);

  const item = classroom;

  return (
    <Theme>
      <Background2
        blue={
          <ContainerB>
            <Logo source={require('../../assets/logo.png')} />
            <UserContatiner>
              <Icon source={require('../../assets/user_blue.png')} />
            </UserContatiner>
            <CustomTextContainer white smallMedium marginTop={{value: '14%'}}>
              {teacher &&
                `${teacher.Student.User.firstName} ${teacher.Student.User.lastName}`}
            </CustomTextContainer>
          </ContainerB>
        }
        white={
          <ContainerW>
            <CustomTextContainer
              black
              smallMedium
              marginTop={{value: '2%'}}
              marginBot={{value: '1%'}}>
              Disciplina
            </CustomTextContainer>
            <RedContainerText medium>
              {teacher && teacher.graduation_area}
            </RedContainerText>
            <CustomTextContainer
              black
              smallMedium
              marginTop={{value: '2%'}}
              marginBot={{value: '1%'}}>
              Formação
            </CustomTextContainer>
            <RedContainerText medium>
              {teacher && gradeResolver(teacher.Student.grade)}
            </RedContainerText>
            <CustomTextContainer
              black
              smallMedium
              marginTop={{value: '2%'}}
              marginBot={{value: '1%'}}>
              Universidade
            </CustomTextContainer>
            <RedContainerText medium>
              {teacher && teacher.Student.institution}
            </RedContainerText>
            <CustomTextContainer
              black
              smallMedium
              marginTop={{value: '2%'}}
              marginBot={{value: '1%'}}>
              Especialidade
            </CustomTextContainer>
            <RedContainerText>
              Ja fiz curso para lecionar para crianças dentro do espectro de
              deficit de atençao.
            </RedContainerText>
            <ButtonContainer>
              <ComplainButton
                testID="ComplainButton"
                onPress={async () => {
                  await getStudent(route.params.selectedId);
                  navigation.navigate('FeedbackTeacher', {item});
                }}>
                <CustomText white bigSmall>
                  Reportar
                </CustomText>
              </ComplainButton>
              <ContinueContainer
                testID="ContinueButton"
                onPress={async () => {
                  await createClass(teacher.id);
                  navigation.navigate('Home');
                }}
              />
            </ButtonContainer>
          </ContainerW>
        }
      />
    </Theme>
  );
}
