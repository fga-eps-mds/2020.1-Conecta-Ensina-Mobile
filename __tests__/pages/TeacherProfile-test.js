import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import TeacherProfile from '../../src/pages/TeacherProfile';
import {ClassroomContext} from '../../src/contexts/classroom';
import {StudentContext} from '../../src/contexts/student';
import {TeacherContext} from '../../src/contexts/teacher';

jest.mock('../../src/contexts/classroom.js');

describe('Testing teacher profile page', () => {
  const createClass = jest.fn();
  const readClass = jest.fn();
  const getStudent = jest.fn();
  const navigation = {
    navigate: jest.fn(),
  };

  const teacher = {
    id: '',
    user: {
      firstName: '',
      lastName: '',
    },
    teacher: {
      graduation_area: '',
    },
    student: {
      grade: '',
      institution: '',
    },
  };
  const route = {
    params: {
      selectedId: ' ',
    },
  };
  test('Should renders page', () => {
    const tree = render(<TeacherProfile route={route} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  test("Should press 'Continue' button", async () => {
    const {getByTestId} = render(
      <StudentContext.Provider value={{getStudent}}>
        <TeacherContext.Provider value={{teacher}}>
          <ClassroomContext.Provider value={{createClass, readClass}}>
            <TeacherProfile navigation={navigation} route={route} />
          </ClassroomContext.Provider>
        </TeacherContext.Provider>
      </StudentContext.Provider>,
    );
    const button = getByTestId('ContinueButton');

    await fireEvent.press(button);
  });
  test("Should press 'Complain Button'", async () => {
    const {getByTestId} = render(
      <StudentContext.Provider value={{getStudent}}>
        <ClassroomContext.Provider value={{createClass}}>
          <TeacherProfile navigation={navigation} route={route} />
        </ClassroomContext.Provider>
      </StudentContext.Provider>,
    );

    const button = getByTestId('ComplainButton');

    await fireEvent.press(button);
  });
});
