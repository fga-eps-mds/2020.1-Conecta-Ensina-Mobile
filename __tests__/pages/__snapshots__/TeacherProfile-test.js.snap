// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing teacher profile page Should renders page 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#184E79",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#184E79",
          "flexBasis": 0,
          "flexGrow": 0.7,
          "flexShrink": 1,
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "marginBottom": "8.4%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/logo.png",
          }
        }
        style={
          Array [
            Object {
              "height": 120,
              "width": 120,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderRadius": 100,
              "height": 58,
              "justifyContent": "center",
              "marginTop": "5.5%",
              "width": 58,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/user_blue.png",
            }
          }
          style={
            Array [
              Object {
                "height": "63%",
                "width": "52%",
              },
            ]
          }
        />
      </View>
      <View
        marginTop={
          Object {
            "value": "14%",
          }
        }
        style={
          Array [
            Object {
              "marginBottom": 0,
              "marginTop": "14%",
            },
          ]
        }
      >
        <Text
          smallMedium={true}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Roboto",
                "fontSize": 20,
                "textAlign": "center",
              },
            ]
          }
          white={true}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <View
        marginBot={
          Object {
            "value": "1%",
          }
        }
        marginTop={
          Object {
            "value": "2%",
          }
        }
        style={
          Array [
            Object {
              "marginBottom": "1%",
              "marginTop": "2%",
            },
          ]
        }
      >
        <Text
          black={true}
          smallMedium={true}
          style={
            Array [
              Object {
                "color": "black",
                "fontFamily": "Roboto",
                "fontSize": 20,
                "textAlign": "center",
              },
            ]
          }
        >
          Disciplina
        </Text>
      </View>
      <View
        medium={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#E46270",
              "borderRadius": 15,
              "height": 40,
              "justifyContent": "center",
              "textAlign": "center",
              "width": 196,
            },
          ]
        }
      >
        <Text
          small={true}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Roboto",
                "fontSize": 14,
                "textAlign": "center",
              },
            ]
          }
          white={true}
        />
      </View>
      <View
        marginBot={
          Object {
            "value": "1%",
          }
        }
        marginTop={
          Object {
            "value": "2%",
          }
        }
        style={
          Array [
            Object {
              "marginBottom": "1%",
              "marginTop": "2%",
            },
          ]
        }
      >
        <Text
          black={true}
          smallMedium={true}
          style={
            Array [
              Object {
                "color": "black",
                "fontFamily": "Roboto",
                "fontSize": 20,
                "textAlign": "center",
              },
            ]
          }
        >
          Formação
        </Text>
      </View>
      <View
        medium={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#E46270",
              "borderRadius": 15,
              "height": 40,
              "justifyContent": "center",
              "textAlign": "center",
              "width": 196,
            },
          ]
        }
      >
        <Text
          small={true}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Roboto",
                "fontSize": 14,
                "textAlign": "center",
              },
            ]
          }
          white={true}
        >
          Invalid Grade
        </Text>
      </View>
      <View
        marginBot={
          Object {
            "value": "1%",
          }
        }
        marginTop={
          Object {
            "value": "2%",
          }
        }
        style={
          Array [
            Object {
              "marginBottom": "1%",
              "marginTop": "2%",
            },
          ]
        }
      >
        <Text
          black={true}
          smallMedium={true}
          style={
            Array [
              Object {
                "color": "black",
                "fontFamily": "Roboto",
                "fontSize": 20,
                "textAlign": "center",
              },
            ]
          }
        >
          Universidade
        </Text>
      </View>
      <View
        medium={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#E46270",
              "borderRadius": 15,
              "height": 40,
              "justifyContent": "center",
              "textAlign": "center",
              "width": 196,
            },
          ]
        }
      >
        <Text
          small={true}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Roboto",
                "fontSize": 14,
                "textAlign": "center",
              },
            ]
          }
          white={true}
        />
      </View>
      <View
        marginBot={
          Object {
            "value": "1%",
          }
        }
        marginTop={
          Object {
            "value": "2%",
          }
        }
        style={
          Array [
            Object {
              "marginBottom": "1%",
              "marginTop": "2%",
            },
          ]
        }
      >
        <Text
          black={true}
          smallMedium={true}
          style={
            Array [
              Object {
                "color": "black",
                "fontFamily": "Roboto",
                "fontSize": 20,
                "textAlign": "center",
              },
            ]
          }
        >
          Especialidade
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#E46270",
              "borderRadius": 15,
              "height": 60,
              "justifyContent": "center",
              "textAlign": "center",
              "width": 280,
            },
          ]
        }
      >
        <Text
          small={true}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Roboto",
                "fontSize": 14,
                "textAlign": "center",
              },
            ]
          }
          white={true}
        >
          Ja fiz curso para lecionar para crianças dentro do espectro de deficit de atençao.
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginTop": 0,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#297FB8",
              "borderRadius": 8,
              "height": 31,
              "justifyContent": "center",
              "opacity": 1,
              "width": 138,
            }
          }
          testID="teacherProfileButton"
        >
          <Text
            bigSmall={true}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 16,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          >
            Continuar
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
